### PR TITLE
Correctly sync deployment image

### PR DESCRIPTION
**Pull request by spadgett**
Simplify `SyncDeployment` and use deployment annotations to force a rollout when the console image changes.

@benjaminapetersen @enj

### CODE REVIEW

File: pkg/console/subresource/deployment/deployment_test.go [hunk 1]
```diff
@@ -180,7 +184,7 @@ func TestStub(t *testing.T) {
 				},
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      api.OpenShiftConsoleName,
-					Namespace: api.OpenShiftConsoleName,
+					Namespace: api.OpenShiftConsoleNamespace,
 					Labels: map[string]string{
```
Comment (jhadvig) [inline pkg/console/subresource/deployment/deployment_test.go]: I know it not new but can we use the `SharedLabels()` here and in other places?

File: pkg/console/subresource/route/route_test.go [hunk 2]
```diff
@@ -89,7 +90,7 @@ func TestStub(t *testing.T) {
 					DeletionTimestamp:          nil,
 					DeletionGracePeriodSeconds: nil,
 					Labels:          map[string]string{"app": api.OpenShiftConsoleName},
```
Comment (jhadvig) [inline pkg/console/subresource/route/route_test.go]: Also here?

File: pkg/console/subresource/secret/secret_test.go [hunk 3]
```diff
@@ -35,9 +35,10 @@ func TestDefaultSecret(t *testing.T) {
 			want: &corev1.Secret{
 				TypeMeta: v1.TypeMeta{},
 				ObjectMeta: v1.ObjectMeta{
-					Name:      deployment.ConsoleOauthConfigName,
-					Namespace: api.OpenShiftConsoleName,
-					Labels:    map[string]string{"app": api.OpenShiftConsoleName},
+					Name:        deployment.ConsoleOauthConfigName,
+					Namespace:   api.OpenShiftConsoleNamespace,
+					Labels:      map[string]string{"app": api.OpenShiftConsoleName},
```
Comment (jhadvig) [inline pkg/console/subresource/secret/secret_test.go]: Also here ?

File: pkg/console/subresource/secret/secret_test.go [hunk 4]
```diff
@@ -64,9 +65,10 @@ func TestStub(t *testing.T) {
...
```
Comment (jhadvig) [inline pkg/console/subresource/secret/secret_test.go]: Also here ?

File: pkg/console/subresource/util/util_test.go [hunk 5]
```diff
@@ -57,9 +57,10 @@ func TestSharedMeta(t *testing.T) {
 		{
 			name: "Test shared metadata",
 			want: v1.ObjectMeta{
-				Name:      api.OpenShiftConsoleName,
-				Namespace: api.OpenShiftConsoleName,
-				Labels:    map[string]string{"app": api.OpenShiftConsoleName},
+				Name:        api.OpenShiftConsoleName,
+				Namespace:   api.OpenShiftConsoleNamespace,
+				Labels:      map[string]string{"app": api.OpenShiftConsoleName},
```
Comment (jhadvig) [inline pkg/console/subresource/util/util_test.go]: Also here?

**Comment by spadgett** [comment]
@jhadvig I think for the unit tests we specifically *don't* want to use `SharedLabels()`. This way we test that `SharedLabels()` is generating expected values.

**Comment by jhadvig** [comment]
ohh.. if the `SharedLabels()` are just for assertion then scratch my comments.

**Comment by benjaminapetersen** [comment]
Still WIP?  Tests pass, def simpler.
We dont have test coverage to ensure a new deployment on changes.  I should add that to the todo list.

**Comment by spadgett** [comment]
@jhadvig @benjaminapetersen thanks, this is ready now

**Comment by benjaminapetersen** [comment]
/retest

not much happening with statuses.

**Comment by benjaminapetersen** [comment]
lgtm if ci gets happy...

File: pkg/console/operator/sync_v400.go [hunk 6]
```diff
@@ -114,44 +113,14 @@ func sync_v400(co *consoleOperator, consoleConfig *v1alpha1.Console) (*v1alpha1.
...
-	// otherwise, we may need to update or force a rollout
-	if deploymentsub.ResourceVersionsChanged(existingDeployment, cm, serviceCAConfigMap, sec) {
-		toUpdate := deploymentsub.UpdateResourceVersions(existingDeployment, cm, serviceCAConfigMap, sec)
-		updatedDeployment, depChanged, updateErr := resourceapply.ApplyDeployment(co.deploymentClient, recorder, toUpdate, deploymentGeneration, true)
-		if updateErr != nil {
-			logrus.Errorf("%q: %v \n", "deployment", updateErr)
-			return nil, false, updateErr
-		}
-		return updatedDeployment, depChanged, nil
+	expectedGeneration := getDeploymentGeneration(co)
```
Comment (benjaminapetersen) [inline pkg/console/operator/sync_v400.go]: So we still have work to do here, with generation, per our conversation. Should we comment to track, or assume a fast follow?

Comment (spadgett) [reply pkg/console/operator/sync_v400.go]: Let's open a JIRA issue

**Pull request merged**